lib/theme_check/position_helper.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+# Note: Everything is 0-indexed here.
+
+module ThemeCheck
+  module PositionHelper
+    def from_row_column_to_index(content, row, col)
+      return 0 unless content.is_a?(String) && !content.empty?
+      return 0 unless row.is_a?(Integer) && col.is_a?(Integer)
+      i = 0
+      result = 0
+      safe_row = bounded(0, row, content.lines.size - 1)
+      lines = content.lines
+      line_size = lines[i].size
+      while i < safe_row
+        result += line_size
+        i += 1
+        line_size = lines[i].size
+      end
+      result += bounded(0, col, line_size - 1)
+      result
+    end
+
+    def from_index_to_row_column(content, index)
+      return [0, 0] unless content.is_a?(String) && !content.empty?
+      return [0, 0] unless index.is_a?(Integer)
+      safe_index = bounded(0, index, content.size - 1)
+      lines = content[0..safe_index].lines
+      row = lines.size - 1
+      col = lines.last.size - 1
+      [row, col]
+    end
+
+    def bounded(a, x, b)
+      [a, [x, b].min].max
+    end
```

Comment:
Suggestion: you could use a range here:

```suggestion
    def bounded(a, range)
      [range.begin, [x, range.end].min].max
    end
```

`bounded(x, 1..3)`

Reply:
Right but I think it would make the calls slightly more difficult to comprehend:

I find the following slightly less cute:
```
bounded(line_number - 1, 0..content.lines.sizes - 1)

# vs

bounded(0, line_number - 1, content.lines.sizes - 1)
```

In my mind it's equivalent to

```
a <= x <= b
```